tensorflow/compiler/tf2xla/xla_op_registry.cc

```diff
@@ -174,6 +175,23 @@ XlaOpRegistry::~XlaOpRegistry() = default;
   }();
   (void)registration_init;
 
+  // Register GPU JIT devices for NextPluggableDevice if its jit_device_type is
+  // `XLA_GPU_JIT`.
+  if (DeviceFactory::IsPluggableDevice(device_name)) {
+    mutex_lock lock(registry.mutex_);
+    NextPluggableDeviceFactory* device_factory =
+        dynamic_cast<NextPluggableDeviceFactory*>(
+            DeviceFactory::GetFactory(device_name));
+    if (device_factory != nullptr &&
+        device_factory->jit_device_type() == DeviceType(DEVICE_GPU_XLA_JIT)) {
+      DeviceRegistration& registration =
+          registry.compilation_devices_[device_name];
+      registration.compilation_device_name = DEVICE_GPU_XLA_JIT;
+      registration.autoclustering_policy =
```

Comment:
does it register each time GetCompilationDevice is called?

Reply:
fixed.